### PR TITLE
Reader: move list streams to Redux

### LINKED ISSUE #4166: Reader: newly followed or unfollowed lists don't change in sidebar until after refresh
Since Redux-ing the Reader lists in the sidebar (#4036), following a new list does not result in it being added to the sidebar immediately. User must refresh the page for it to appear.

We need to add the new list to the Redux state whilst maintaining the existing Flux actions (the list follow button and list stream header rely on the Flux store - see #4165).

### PR DESCRIPTION
**Pull request by bluefuton**
The current user's Reader lists are currently stored in Flux stores, and also partly in Redux state since #4036.

This PR moves list streams over to Redux completely, including following state for public lists.

There are no functional changes, so list streams should function exactly as they currently do on production - it's just the underpinnings that are changing.

Fixes #4166.

### To test

Visit a list stream (e.g. http://calypso.localhost:3000/read/list/blowery/comics) and ensure that:
- the list details (title and description) are correctly displayed
- for public lists not owned by you - following and unfollowing the list works as expected
- any changes in followed and unfollowed lists are reflected in the sidebar

screen shot 2016-03-30 at 23 15 26

### CODE REVIEW

**Comment by bluefuton** [comment]
Todo:
- [ ] move list retrieval to Redux (new query component?)
- [ ] move follow to Redux
- [ ] move unfollow to Redux

**Comment by bluefuton** [comment]
One known issue: loading a non-existent list does not result in a 404. This mirrors the current behaviour on production, so I'd like to fix it in a separate PR. There's an existing issue for this: #3681.

**Comment by bluefuton** [comment]
We may want to optimistically update the following state before the API response is received.

**Comment by bluefuton** [comment]
Ready for a first :eyes: @blowery :)

**Comment by blowery** [comment]
@bluefuton be warned, rebased and force pushed this one

**Comment by blowery** [comment]
One odd issue, when I change a list name of a list I own, the name doesn't update in the sidebar until I leave the settings menu. I would expect it to update on save.

**Comment by bluefuton** [comment]
> One odd issue, when I change a list name of a list I own, the name doesn't update in the sidebar until I leave the settings menu. I would expect it to update on save.

The new list title/description edit page is still running off the Flux stores at this point (it'll be shifted across in #4144). Users in any environment except dev will still see the old edit screen and sidebar, so they shouldn't run into this.

File: client/reader/controller.js [hunk 1]
```diff
@@ -411,23 +411,23 @@ module.exports = {
...
+					slug: context.params.list,
+					setPageTitle: setPageTitle,
+					trackScrollPage: trackScrollPage.bind(
+						null,
+						basePath,
+						fullAnalyticsPageTitle,
+						analyticsPageTitle,
+						mcKey
+					),
+					onUpdatesShown: trackUpdatesLoaded.bind( null, mcKey )
```
Comment (blowery) [inline client/reader/controller.js]: prolly Shown?

File: client/state/reader/lists/test/actions.js [hunk 2]
```diff
@@ -74,4 +102,46 @@ describe( 'actions', () => {
 			} );
 		} );
 	} );
+
+	describe( '#followList()', () => {
+		before( () => {
+			nock( 'https://public-api.wordpress.com:443' )
+				.persist()
```
Comment (blowery) [inline client/state/reader/lists/test/actions.js]: don't need all the persists. Probably should use `useNock` from helpers as we chatted about at the meetup.

**Comment by bluefuton** [comment]
- [ ] fix double-encoding of slugs in follow/unfollow request

File: client/state/reader/lists/test/actions.js (hunk 2 above)
Comment (bluefuton) [inline client/state/reader/lists/test/actions.js]: :+1:

**Comment by blowery** [comment]
GTG :ship: :rocket:

**Pull request merged**